dev/tests/integration/testsuite/Magento/Payment/Helper/DataTest.php

```diff
@@ -21,4 +23,36 @@ public function testGetInfoBlock()
         $result = $helper->getInfoBlock($paymentInfo);
         $this->assertInstanceOf(\Magento\OfflinePayments\Block\Info\Checkmo::class, $result);
     }
+
+    /**
+     * Checking if payment available in getPaymentMethodList() with set active to 0.
+     *
+     * @magentoConfigFixture default_store payment/checkmo/active 0
+     */
+    public function testGetPaymentMethodListWithInActivePayments()
+    {
+        /** @var \Magento\Payment\Helper\Data $helper */
+        $helper = Bootstrap::getObjectManager()->get(\Magento\Payment\Helper\Data::class);
+        $paymentMethodsList =  $helper->getPaymentMethodList();
+        $this->assertArrayHasKey('cashondelivery',$paymentMethodsList);
+        $this->assertArrayHasKey('checkmo',$paymentMethodsList);
+    }
+
+    /**
+     * Checking different title for payment method.
+     *
+     * @magentoConfigFixture default_store payment/checkmo/title New Title
+     */
+    public function testGetPaymentMethodListWithDifferentTitles()
+    {
+        $helper = Bootstrap::getObjectManager()->get(\Magento\Payment\Helper\Data::class);
+        $storedTitle = $helper->getMethodInstance('checkmo')->getConfigData('title', 'default');
+
+        foreach ($helper->getPaymentMethodList() as $code => $title) {
+            if ($code == 'checkmo') {
+                $this->assertEquals($storedTitle, $title);
```

Comment:
Please avoid logic in test. Why we cannot assert `$payments['checkmo']` with `assertSame` instead?

Reply:
yes i agree with you saw this after i push commit . thanks :+1: